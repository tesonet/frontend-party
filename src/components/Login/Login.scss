.login {
  height         : 100%;
  width          : 100%;
  display        : flex;
  align-items    : center;
  background     : url('/static/images/bg.jpg') no-repeat center center fixed;
  background-size: cover;
  padding        : 0 2em;

  &__container {
    width: 100%;

    &-centered {
      display: block;
      margin : 0 auto;
    }
  }

  &__logo {
    display: block;
    margin : 0 auto 4rem;
  }
}