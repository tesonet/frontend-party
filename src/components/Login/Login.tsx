import React from "react";

const Login: React.FC = () => <div>Login</div>;

export default Login;
