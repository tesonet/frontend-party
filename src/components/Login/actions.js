import api from "../../service/api";
import {useDispatch, useSelector} from "react-redux";
import actionList from './../../common/actionList'
import { validateAndAuthenticate } from "../../common/common";

export default function (history) {
    const dispatch = useDispatch()
    const credentials = useSelector(state => state.generalReducer.credentials)
    const failedLogin = useSelector(state => state.generalReducer.failedLogin)

    return {
        handleUsername:  (e) => {
            dispatch({type: 'SET_CREDENTIALS', payload: {name: e.currentTarget.value, pass: credentials.pass}})
        },
        handlePassword: (e) => {
            dispatch({type: 'SET_CREDENTIALS', payload: {name: credentials.name, pass: e.currentTarget.value}})
        },
        handleLogin: async function (e) {
            e.preventDefault()
            dispatch(actionList.enablePreloader())
            let response = await api.login(credentials.name, credentials.pass)
            dispatch(actionList.disablePreloader())
            validateAndAuthenticate(response, history, dispatch);
        },
        failedLogin
    }
}