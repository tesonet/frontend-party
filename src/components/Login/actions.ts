import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE
} from '../../redux/actionsConst';

/* Action payloads interfaces */
export interface LoginCredentials {
  username: string;
  password: string;
}

export interface Error {
  error: string;
}

/* Action interfaces */
export interface LoginRequestAction {
  type: typeof LOGIN_REQUEST;
  data: LoginCredentials;
}

export interface LoginSuccessAction {
  type: typeof LOGIN_SUCCESS;
}

export interface LoginFailureAction {
  type: typeof LOGIN_FAILURE;
  data: Error
}

export type LoginActionTypes =
  | LoginRequestAction
  | LoginSuccessAction
  | LoginFailureAction;

/* Actions */
export const loginRequest= (data: LoginCredentials) => ({
  type: LOGIN_REQUEST,
  data
});

export const loginSuccess = () => ({
  type: LOGIN_SUCCESS
});

export const loginFailure = (data: Error) => ({
  type: LOGIN_FAILURE,
  data
});
