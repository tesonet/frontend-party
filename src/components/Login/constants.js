import Yup from 'yup';

import { FORM_AUTH } from '../../constants/forms';

// Auth form configuration
export const FORM_AUTH_CONFIG = {
  mapPropsToValues: () => ({
    username: '',
    password: '',
  }),
  validationSchema: Yup.object().shape({
    username: Yup.string().required('Please enter the username.'),
    password: Yup.string().required('Please enter the password.'),
  }),
  handleSubmit: (values, { setSubmitting, props: { login } }) => {
    login(values.username, values.password);
    setSubmitting(false);
  },
  displayName: FORM_AUTH,
};
