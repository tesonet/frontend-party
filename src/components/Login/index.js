export { default as LoginBackground } from './LoginBackground';
export { default as LoginLogo } from './LoginLogo';
export { default as LoginForm } from './LoginForm';
export { default as LoginInput } from './LoginInput';
export { default as LoginButton } from './LoginButton';
export { default as LoginError } from './LoginError';
export { default as LoginRequiredError } from './LoginRequiredError';
