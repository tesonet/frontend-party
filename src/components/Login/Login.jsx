import React, { memo } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import LoginForm from './LoginForm';
import actions from '../../actions/authActions';
import testioLogo from '../../assets/logo-testio.svg';

const Login = () => {
    const history = useHistory();
    const dispatch = useDispatch();
    const onSubmit = (values) => dispatch(actions.login(values, history));

    return (
        <div className="login-page">
            <img src={testioLogo} alt="" className="login-logo" />
            <LoginForm onSubmit={onSubmit} />
        </div>
    );
};

export default memo(Login);
