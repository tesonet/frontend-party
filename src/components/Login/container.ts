import { connect } from 'react-redux';
import { loginRequest } from './actions';
import { Login as Component } from './Login';

const mapStateToProps = ({ login }: any) => ({
  authLoading: login.authLoading,
  error: login.error
});

const mapDispatchToProps = {
  loginRequest
};

export const Login = connect(mapStateToProps, mapDispatchToProps)(Component);
