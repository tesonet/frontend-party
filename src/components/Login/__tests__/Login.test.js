import React from 'react';
import { shallow } from 'enzyme';

import Login from '../Login';

const withFormik = jest.fn(); // eslint-disable-line no-unused-vars

describe('Login component', () => {
  const props = {
    store: {
      getState: () => ({
        auth: {
          token: null,
          isLoading: false,
          error: null,
        },
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    },
    values: {},
    touched: {},
    errors: {},
    isSubmitting: false,
    handleChange: jest.fn(),
    handleBlur: jest.fn(),
    handleSubmit: jest.fn(),
    push: jest.fn(),
    errorDismiss: jest.fn(),
    isLoading: false,
    token: null,
    error: null,
  };

  it('should render the Login component.', () => {
    const renderedComponent = shallow(<Login {...props} />).dive();

    expect(renderedComponent).toMatchSnapshot();
  });
});
