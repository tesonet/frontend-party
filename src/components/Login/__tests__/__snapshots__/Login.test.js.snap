// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render the Login component. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <C
    error={null}
    errorDismiss={[Function]}
    errors={Object {}}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleSubmit={[Function]}
    isLoading={false}
    isSubmitting={false}
    login={[Function]}
    push={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": undefined,
          }
    }
    token={null}
    touched={Object {}}
    values={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "displayName": "FORM_AUTH",
      "enableReinitialize": false,
      "error": null,
      "errorDismiss": [Function],
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "password": "",
        "username": "",
      },
      "isInitialValid": false,
      "isLoading": false,
      "isSubmitting": false,
      "login": [Function],
      "onSubmit": [Function],
      "push": [Function],
      "render": [Function],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": undefined,
      },
      "token": null,
      "touched": Object {},
      "validate": undefined,
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": [Function],
      "values": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "displayName": "FORM_AUTH",
        "enableReinitialize": false,
        "error": null,
        "errorDismiss": [Function],
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "password": "",
          "username": "",
        },
        "isInitialValid": false,
        "isLoading": false,
        "isSubmitting": false,
        "login": [Function],
        "onSubmit": [Function],
        "push": [Function],
        "render": [Function],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": undefined,
        },
        "token": null,
        "touched": Object {},
        "validate": undefined,
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
        "values": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
